src/utils/utils.cpp

```diff
@@ -192,48 +206,115 @@ void PrintValue(std::ostream &os, const mg_path *path) {
   }
 }
 
-void PrintValue(std::ostream &os, const mg_value *value) {
-  switch (mg_value_get_type(value)) {
-    case MG_VALUE_TYPE_NULL:
-      os << "Null";
-      return;
-    case MG_VALUE_TYPE_BOOL:
-      os << (mg_value_bool(value) ? "true" : "false");
-      return;
-    case MG_VALUE_TYPE_INTEGER:
-      os << mg_value_integer(value);
-      return;
-    case MG_VALUE_TYPE_FLOAT:
-      os << mg_value_float(value);
-      return;
-    case MG_VALUE_TYPE_STRING:
-      PrintValue(os, mg_value_string(value));
-      return;
-    case MG_VALUE_TYPE_LIST:
-      PrintValue(os, mg_value_list(value));
-      return;
-    case MG_VALUE_TYPE_MAP:
-      PrintValue(os, mg_value_map(value));
-      return;
-    case MG_VALUE_TYPE_NODE:
-      PrintValue(os, mg_value_node(value));
-      return;
-    case MG_VALUE_TYPE_RELATIONSHIP:
-      PrintValue(os, mg_value_relationship(value));
-      return;
-    case MG_VALUE_TYPE_UNBOUND_RELATIONSHIP:
-      PrintValue(os, mg_value_unbound_relationship(value));
-      return;
-    case MG_VALUE_TYPE_PATH:
-      PrintValue(os, mg_value_path(value));
-      return;
-    default:
-      os << "{unknown value}";
-      break;
-  }
+void PrintValue(std::ostream &os, const mg_date *date) {
+  const auto days = date::days(mg_date_days(date));
+  const auto sys_days = date::sys_days(days);
+  const auto ymd = date::year_month_day(sys_days);
+  os << ymd;
+}
+
+void PrintValue(std::ostream &os, const mg_local_time *local_time) {
+  const auto nanos =
+      std::chrono::nanoseconds(mg_local_time_nanoseconds(local_time));
+  const auto hms = date::hh_mm_ss(nanos);
+  os << hms;
+}
+
+void PrintValue(std::ostream &os, const mg_local_date_time *local_date_time) {
+  const auto seconds =
+      std::chrono::seconds(mg_local_date_time_seconds(local_date_time));
+  const auto days = std::chrono::duration_cast<date::days>(seconds);
+  const auto date = mg_date_make(days.count());
+
+  const auto nanoseconds =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(seconds) -
+      std::chrono::duration_cast<std::chrono::nanoseconds>(days);
+  const auto time = mg_local_time_make(nanoseconds.count());
+
+  PrintValue(os, date);
+  os << " ";
+  PrintValue(os, time);
 }
 
-}  // namespace utils
+void PrintValue(std::ostream &os, const mg_duration *duration) {
+  // const auto months = date::months(mg_duration_months(duration));
+  const auto days = date::days(mg_duration_days(duration));
+  const auto seconds = std::chrono::seconds(mg_duration_seconds(duration));
+  const auto nanoseconds =
+      std::chrono::nanoseconds(mg_duration_nanoseconds(duration));
+
+  const auto time = std::chrono::duration_cast<std::chrono::microseconds>(
+      seconds + nanoseconds);
+
+  const auto hh = std::chrono::duration_cast<std::chrono::hours>(time);
```

Comment:
We have a utility function internally to do this behavior. You can copy/paste it and use that ;)

Reply:
I saw how it works but didn't want to cp the function wince its needed only for duration